hollow/src/main/java/com/netflix/hollow/api/producer/enforcer/AbstractSingleProducerEnforcer.java

```diff
@@ -27,6 +29,8 @@
     private boolean wasPrimary = false;
     private final Logger logger = Logger.getLogger(AbstractSingleProducerEnforcer.class.getName());
 
+    private volatile Lock lock = new ReentrantLock();
```

Comment:
I think this would be better as `final` and not `volatile` since it's never reassigned?

Reply:
Indeed